### PR TITLE
Merchant

### PR DESCRIPTION
**Pull request by KaraJoHo**
Merging Merchant branch to master branch

### CODE REVIEW

**Review by beddings81 — COMMENTED** [review]
Great job!

File: lib/merchant.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Merchant
+  attr_accessor :id, :name
+  def initialize(attributes)
+    @id = attributes[:id]
+    @name =attributes[:name]
```
Comment (Pocketzs) [inline lib/merchant.rb]: Add space after  = sign

Comment (Pocketzs) [inline lib/merchant.rb]: Add space after = sign on this line

File: lib/merchant_repository.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  def find_by_id(id)
+    @all.find do |merchant|
+      merchant.id == id
+    end
+  end
+
+  def find_by_name(name)
+    @all.find do |merchant|
+      name.casecmp(merchant.name) == 0
```
Comment (Pocketzs) [inline lib/merchant_repository.rb]: Change `name.casecmp(merchant.name) == 0` to `name.casecmp?(merchant.name)`

File: lib/merchant_repository.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end
+
+  def find_all_by_name(name)
+    @all.find_all do |merchant|
+      name.casecmp(merchant.name) == 0
```
Comment (Pocketzs) [inline lib/merchant_repository.rb]: same

File: lib/merchant_repository.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end
+
+  def create(attributes)
+    new = Merchant.new(attributes)
+    new.id = @all.max_by do |merchant|
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply lib/merchant_repository.rb:33]: Don't worry about this

File: lib/merchant_repository.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      merchant.id
+    end.id + 1
+    new
+  end
+
+  def update(id, attributes)
+     update_merchant = find_by_id(id)
```
Comment (Pocketzs) [inline lib/merchant_repository.rb]: Fix your VS code or Atom settings to indent only 2 spaces

**Review by Pocketzs — CHANGES_REQUESTED** [review]
Fix the following commented changes

**Review by Kledin85 — CHANGES_REQUESTED** [review]
fix hound comments

**Review by Pocketzs — APPROVED** [review]

**Review by Kledin85 — APPROVED** [review]

**Pull request merged**